### ISSUE TITLE
macOS: Build command fails

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#5740]
**Describe the bug**
The building of macOS fails with the following error:
```
CompileC /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_38_cxx.o /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.dir/Unity/unity_38_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'AzToolsFramework' from project 'O3DE')
    cd /Users/mac05/o3de
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target x86_64-apple-macos11.0 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O2 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR\=\"profile\" -D_HAS_EXCEPTIONS\=0 -D_PROFILE -DAZ_PROFILE_BUILD\=1 -DNDEBUG -DAZ_ENABLE_TRACING -DAZ_ENABLE_DEBUG_TOOLS -DAZ_BUILD_CONFIGURATION_TYPE\=\"profile\" -DAPPLE -DMAC -D__APPLE__ -DDARWIN -DQT_CORE_LIB -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_WIDGETS_LIB -DQT_SVG_LIB -DQT_XML_LIB -DQT_MACEXTRAS_LIB -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/mac05/o3de/osx_build/lib/profile/include -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include_profile -I/Users/mac05/o3de/Code/Framework/AzToolsFramework -I/Users/mac05/o3de/Code/Framework/AzToolsFramework/AzToolsFramework -I/Users/mac05/o3de/Code/Framework/AzToolsFramework/Platform/Mac -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include/AzToolsFramework -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include/Platform/Mac -I/Users/mac05/o3de/Code/Framework/AzCore -I/Users/mac05/o3de/Code/Framework/AzCore/Platform/Mac -I/Users/mac05/o3de/Code/Framework/AzCore/Platform/Common -I/Users/mac05/o3de/Code/Framework/AzFramework -I/Users/mac05/o3de/Code/Framework/AzFramework/Platform/Mac -I/Users/mac05/o3de/Code/Framework/GridMate -I/Users/mac05/o3de/Code/Framework/GridMate/Platform/Mac -I/Users/mac05/o3de/Code/Framework/AzQtComponents -I/Users/mac05/o3de/Code/Framework/AzQtComponents/Platform/Mac -isystem /Users/mac05/o3de/3rdParty/packages/SQLite-3.32.2-rev3-multiplatform/SQLite -isystem /Users/mac05/o3de/3rdParty/packages/Lua-5.3.5-rev6-mac/Lua/include -isystem /Users/mac05/o3de/3rdParty/packages/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include -isystem /Users/mac05/o3de/3rdParty/packages/RapidXML-1.13-rev1-multiplatform/RapidXML/include -isystem /Users/mac05/o3de/3rdParty/packages/zlib-1.2.11-rev5-mac/zlib/include -isystem /Users/mac05/o3de/3rdParty/packages/zstd-1.35-multiplatform/zstd/lib -isystem /Users/mac05/o3de/3rdParty/packages/cityhash-1.1-multiplatform/cityhash/src -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/mkspecs/macx-clang -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/include -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtGui.framework/Headers -isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtGui.framework/Versions/5/Headers/5.15.1 -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtGui.framework/Versions/5/Headers/5.15.1/QtGui -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework/Versions/5/Headers/5.15.1 -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework/Versions/5/Headers/5.15.1/QtCore -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtWidgets.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtWidgets.framework/Versions/5/Headers/5.15.1 -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtWidgets.framework/Versions/5/Headers/5.15.1/QtWidgets -isystem /Users/mac05/o3de/3rdParty/packages/OpenSSL-1.1.1b-rev1-mac/OpenSSL/include -isystem /Users/mac05/o3de/3rdParty/packages/lz4-1.9.3-vcpkg-rev4-mac/lz4/lz4/include -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtSvg.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtXml.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtMacExtras.framework/Headers -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/DerivedSources-normal/x86_64 -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/DerivedSources/x86_64 -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/DerivedSources -F/Users/mac05/o3de/osx_build/lib/profile -F/Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib -fno-exceptions -fvisibility\=hidden -fvisibility-inlines-hidden -Wall -Werror -Wno-inconsistent-missing-override -Wrange-loop-analysis -Wno-unknown-warning-option -Wno-parentheses -Wno-reorder -Wno-switch -Wno-undefined-var-template -fPIC -std\=c++17 -MMD -MT dependencies -MF /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_38_cxx.d --serialize-diagnostics /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_38_cxx.dia -c /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.dir/Unity/unity_38_cxx.cxx -o /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_38_cxx.o
In file included from /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.dir/Unity/unity_38_cxx.cxx:15:
In file included from /Users/mac05/o3de/Code/Framework/AzToolsFramework/Platform/Common/Default/AzToolsFramework/API/PythonLoader_Default.cpp:9:
/Users/mac05/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/API/PythonLoader.h:22:15: error: private field 'm_embeddedLibPythonHandle' is not used [-Werror,-Wunused-private-field]
        void* m_embeddedLibPythonHandle{ nullptr };
              ^
1 error generated.

CompileC /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_37_cxx.o /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.dir/Unity/unity_37_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'AzToolsFramework' from project 'O3DE')
    cd /Users/mac05/o3de
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target x86_64-apple-macos11.0 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O2 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR\=\"profile\" -D_HAS_EXCEPTIONS\=0 -D_PROFILE -DAZ_PROFILE_BUILD\=1 -DNDEBUG -DAZ_ENABLE_TRACING -DAZ_ENABLE_DEBUG_TOOLS -DAZ_BUILD_CONFIGURATION_TYPE\=\"profile\" -DAPPLE -DMAC -D__APPLE__ -DDARWIN -DQT_CORE_LIB -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_WIDGETS_LIB -DQT_SVG_LIB -DQT_XML_LIB -DQT_MACEXTRAS_LIB -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/mac05/o3de/osx_build/lib/profile/include -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include_profile -I/Users/mac05/o3de/Code/Framework/AzToolsFramework -I/Users/mac05/o3de/Code/Framework/AzToolsFramework/AzToolsFramework -I/Users/mac05/o3de/Code/Framework/AzToolsFramework/Platform/Mac -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include/AzToolsFramework -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/include/Platform/Mac -I/Users/mac05/o3de/Code/Framework/AzCore -I/Users/mac05/o3de/Code/Framework/AzCore/Platform/Mac -I/Users/mac05/o3de/Code/Framework/AzCore/Platform/Common -I/Users/mac05/o3de/Code/Framework/AzFramework -I/Users/mac05/o3de/Code/Framework/AzFramework/Platform/Mac -I/Users/mac05/o3de/Code/Framework/GridMate -I/Users/mac05/o3de/Code/Framework/GridMate/Platform/Mac -I/Users/mac05/o3de/Code/Framework/AzQtComponents -I/Users/mac05/o3de/Code/Framework/AzQtComponents/Platform/Mac -isystem /Users/mac05/o3de/3rdParty/packages/SQLite-3.32.2-rev3-multiplatform/SQLite -isystem /Users/mac05/o3de/3rdParty/packages/Lua-5.3.5-rev6-mac/Lua/include -isystem /Users/mac05/o3de/3rdParty/packages/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include -isystem /Users/mac05/o3de/3rdParty/packages/RapidXML-1.13-rev1-multiplatform/RapidXML/include -isystem /Users/mac05/o3de/3rdParty/packages/zlib-1.2.11-rev5-mac/zlib/include -isystem /Users/mac05/o3de/3rdParty/packages/zstd-1.35-multiplatform/zstd/lib -isystem /Users/mac05/o3de/3rdParty/packages/cityhash-1.1-multiplatform/cityhash/src -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/mkspecs/macx-clang -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/include -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtGui.framework/Headers -isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtGui.framework/Versions/5/Headers/5.15.1 -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtGui.framework/Versions/5/Headers/5.15.1/QtGui -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework/Versions/5/Headers/5.15.1 -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework/Versions/5/Headers/5.15.1/QtCore -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtWidgets.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtWidgets.framework/Versions/5/Headers/5.15.1 -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtWidgets.framework/Versions/5/Headers/5.15.1/QtWidgets -isystem /Users/mac05/o3de/3rdParty/packages/OpenSSL-1.1.1b-rev1-mac/OpenSSL/include -isystem /Users/mac05/o3de/3rdParty/packages/lz4-1.9.3-vcpkg-rev4-mac/lz4/lz4/include -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtSvg.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtXml.framework/Headers -isystem /Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtMacExtras.framework/Headers -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/DerivedSources-normal/x86_64 -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/DerivedSources/x86_64 -I/Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/DerivedSources -F/Users/mac05/o3de/osx_build/lib/profile -F/Users/mac05/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib -fno-exceptions -fvisibility\=hidden -fvisibility-inlines-hidden -Wall -Werror -Wno-inconsistent-missing-override -Wrange-loop-analysis -Wno-unknown-warning-option -Wno-parentheses -Wno-reorder -Wno-switch -Wno-undefined-var-template -fPIC -std\=c++17 -MMD -MT dependencies -MF /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_37_cxx.d --serialize-diagnostics /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_37_cxx.dia -c /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.dir/Unity/unity_37_cxx.cxx -o /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_37_cxx.o

** BUILD FAILED **


The following build commands failed:
	CompileC /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.build/Objects-normal/x86_64/unity_38_cxx.o /Users/mac05/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.dir/Unity/unity_38_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'AzToolsFramework' from project 'O3DE')
(1 failure)
```

**Steps to reproduce**
1. Create new or move your preexisting 3rdParty folder containing CMake into o3de folder.
2. Obtain python packages with get_python.sh script located in the o3de/python directory.
3. Create osx_build folder under the o3de location. 
4. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
export LY_PACKAGE_SERVER_URLS=https://d2c171ws20a1rv.cloudfront.net
```
5. Configure and generate the Xcode solution using this command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`
6. Build Mac binaries:
`cmake -E time cmake --build . --target ALL_BUILD --config profile -j 2`

**Expected behavior**
The building completes successfully.

**Actual behavior**
The building fails with an error.

**Found in Branch**
Stabilization/2110 (d222aee)

**Desktop/Device:**
Device: MacBook Pro (15-inch, 2017)
OS: macOS
Version: Big Sur 11.5.1
CPU: Intel Core i7 2,8 GHz Quad-Core
GPU: Radeon Pro 555
Memory: 16GB

**Additional context**
[macOS_build_failed.txt](https://github.com/o3de/o3de/files/7562438/macOS_build_failed.txt)
Building instructions: https://wiki.agscollab.com/pages/viewpage.action?pageId=119004516

**Comment by LB-KacperKapusta** [comment]
It has been verified on Development (239e030) branch that building following the same instructions also fails with in fact different error:
```
CMake Error at /Users/lumberpro9/o3de/osx_build/runtime_dependencies/profile/AzQtComponents.cmake:128 (file):
 file SIZE requested of path that is not readable:
  /Users/lumberpro9/o3de/3rdParty/packages/qt-5.15.2-rev5-mac/qt/lib/QtCore.framework
Call Stack (most recent call first):
 /Users/lumberpro9/o3de/osx_build/runtime_dependencies/profile/AzQtComponents.cmake:195 (ly_copy)
Command PhaseScriptExecution failed with a nonzero exit code
** BUILD FAILED **
The following build commands failed:
	PhaseScriptExecution CMake\ PostBuild\ Rules /Users/lumberpro9/o3de/osx_build/Code/Framework/AzQtComponents/O3DE.build/profile/AzQtComponents.build/Script-E69984AEBCFD9BD4B2054083.sh (in target 'AzQtComponents' from project 'O3DE')
(1 failure)
Elapsed time: 2454 s. (time), 0.049246 s. (clock)
lumberpro9@lumberpro09 osx_build %
```
Please let us know if this can be fixed together or should be logged as separate tickets.

**Comment by LB-KacperKapusta** [comment]
Reopening the issue as it still occurs in Stabilization/2110 (d30fdb7) branch with the same error as described.
Please note that it has been confirmed as fixed in Development (e804fd2) branch and the building completed successfully.

**Comment by amznestebanpapp** [comment]
I'm in the process of merging the fix into stabilization/2110 here: https://github.com/o3de/o3de/pull/5778

**Comment by LB-KacperKapusta** [comment]
Verified the issue as fixed in Stabilization/2110 (a0ab592).
